rpc.go

```diff
@@ -554,6 +559,102 @@ func (c *client) lookupRegion(ctx context.Context,
 	}
 }
 
+func (c *client) lookupAllRegions(ctx context.Context,
+	table []byte) ([]regionInfoAndAddr, error) {
+	var regs []regionInfoAndAddr
+	var reg hrpc.RegionInfo
+	var addr string
+	var err error
+	backoff := backoffStart
+	for {
+		// If it takes longer than regionLookupTimeout, fail so that we can sleep
+		lookupCtx, cancel := context.WithTimeout(ctx, c.regionLookupTimeout)
+		if c.clientType == region.MasterClient {
+			log.WithField("resource", zk.Master).Debug("looking up master")
+
+			addr, err = c.zkLookup(lookupCtx, zk.Master)
+			cancel()
+			reg = c.adminRegionInfo
+			regs = append(regs, regionInfoAndAddr{regionInfo: reg, addr: addr})
+		} else if bytes.Equal(table, metaTableName) {
+			log.WithField("resource", zk.Meta).Debug("looking up region server of hbase:meta")
+
+			addr, err = c.zkLookup(lookupCtx, zk.Meta)
+			cancel()
+			reg = c.metaRegionInfo
+			regs = append(regs, regionInfoAndAddr{regionInfo: reg, addr: addr})
+		} else {
```

Comment:
This code shouldn't be needed I think. You wouldn't use lookupAllRegions to look up the admin or meta regions.